platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheBuildOperations.kt

```diff
@@ -82,17 +83,48 @@ data class LoadResult(val stateFiles: List<File>, val originInvocationId: String
 
 
 private
-object StoreDetails : ConfigurationCacheStoreBuildOperationType.Details
+object WorkGraphStoreDetails : ConfigurationCacheStoreBuildOperationType.Details
 
 
 internal
-data class StoreResult(val stateFiles: List<File>, val storeFailure: Throwable?) : ConfigurationCacheStoreBuildOperationType.Result {
+data class WorkGraphStoreResult(val stateFiles: List<File>, val storeFailure: Throwable?) : ConfigurationCacheStoreBuildOperationType.Result {
     override fun getCacheEntrySize(): Long = stateFiles.asSequence()
         .filter { it.isFile }
         .sumOf { it.length() }
 }
 
 
+internal
+fun BuildOperationRunner.withModelStoreOperation(block: () -> ModelStoreResult): Unit =
+    run(object : RunnableBuildOperation {
+        override fun description(): BuildOperationDescriptor.Builder = BuildOperationDescriptor
+            .displayName("Store model in configuration cache")
+            .progressDisplayName("Storing model in configuration cache")
+
+        override fun run(context: BuildOperationContext) {
+            block().let {
+                context.setResult(it)
+                it.storeFailure?.let { failure -> context.failed(failure) }
+            }
+        }
+    })
+
+
+internal
+fun <T : Any> BuildOperationRunner.withModelLoadOperation(block: () -> T): T =
+    call(object : CallableBuildOperation<T> {
+        override fun description(): BuildOperationDescriptor.Builder = BuildOperationDescriptor
+            .displayName("Load model from configuration cache")
```

Comment:
❓ Do you have some more information which model you are loading? Not sure if that is evident to users.

Reply:
Not really. At this point, all we know is that the last time Gradle was invoked with the same build action, the result was this model -- whatever the user has returned [from `BuildAction.execute`](https://github.com/gradle/gradle/blob/82937b89af179043f34052cd5d6a8818571cca8f/platforms/ide/tooling-api/src/main/java/org/gradle/tooling/BuildAction.java#L37)

I do think we don't have a good name for this model overall, with relation to other things. In code, it's mainly called plainly the "model". But really it's the "result model of the main build action of the TAPI invocation". Though, it's a separate topic.

I would keep it minimal for now, given it's an internal build operation. Later we could add metadata things like the size of this model on disk or the time it took to load it, similarly to the work graph.